exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,768 @@
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, common.EncodeURLValues(marketsPathV3+"trades", params), &resp)
+}
+
+// GetLiquidationOrder get Liquidation Order Interface
+func (e *Exchange) GetLiquidationOrder(ctx context.Context, symbol, direction string, startTime, endTime time.Time, offset, limit uint64) ([]*LiquidationPrice, error) {
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+			return nil, err
+		}
+	}
+	params := url.Values{}
+	if !startTime.IsZero() {
+		params.Set("sTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if !endTime.IsZero() {
+		params.Set("eTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if direction != "" {
+		params.Set("direct", direction)
+	}
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	if offset > 0 {
+		params.Set("from", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp []*LiquidationPrice
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, common.EncodeURLValues(marketsPathV3+"liquidationOrder", params), &resp)
+}
+
+// GetFuturesMarket get the market information of trading pairs in the past 24 hours.
+func (e *Exchange) GetFuturesMarket(ctx context.Context, symbol string) ([]*FuturesTickerDetails, error) {
+	params := url.Values{}
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	var resp []*FuturesTickerDetails
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, common.EncodeURLValues(marketsPathV3+"tickers", params), &resp)
+}
+
+// GetFuturesIndexPrice get the current index price.
+func (e *Exchange) GetFuturesIndexPrice(ctx context.Context, symbol string) (*InstrumentIndexPrice, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	var resp *InstrumentIndexPrice
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, common.EncodeURLValues(marketsPathV3+"indexPrice", params), &resp)
+}
+
+// GetFuturesIndexPrices get the current index price for all instruments
+func (e *Exchange) GetFuturesIndexPrices(ctx context.Context) ([]*InstrumentIndexPrice, error) {
+	var resp []*InstrumentIndexPrice
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, marketsPathV3+"indexPrice", &resp)
+}
+
+// GetIndexPriceComponents get the index price components for a trading pair.
+func (e *Exchange) GetIndexPriceComponents(ctx context.Context, symbol string) (*IndexPriceComponent, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	var resp *IndexPriceComponent
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, fMarketEPL, common.EncodeURLValues(marketsPathV3+"indexPriceComponents", params), &resp)
+}
+
+// GetInstrumentsIndexPriceComponents returns index price components for all trading pairs.
```

Comment:
Comment suggests multiple results are returned but you only return one struct for one symbol

Either this is a single item (requires a param) or you need to update to slice response

Does this endpoint actually exist?
https://api-docs.poloniex.com/v3/futures/api/market/get-index-price-components